src/adsp_config.c

```diff
@@ -1559,13 +1559,35 @@ static int parse_version(toml_table_t *toml, int64_t version[2])
 	return 0;
 }
 
-static inline bool check_config_version(int major, int minor, const int64_t *version)
+struct parser {
```

Comment:
version_parser ? It seems specific for parsing versions only ?

Reply:
no, it's parser for whole configuration file, version is used only to select right parser

Reply:
renamed to `config_parser`